core/src/main/java/com/graphhopper/util/PathSimplification.java

```diff
@@ -98,14 +98,21 @@ public PointList simplify() {
                     int removed = douglasPeucker.simplify(pointList, nonConflictingStart, nonConflictingEnd);
                     if (removed > 0) {
                         for (int i = 0; i < listsToSimplify.size(); i++) {
-                            List pathDetails = listsToSimplify.get(i);
-                            reduceLength(pathDetails, offsets[i], startIntervals[i], endIntervals[i] - removed);
-                            // This is not needed for Instructions, as they don't contain references, but PointLists
-                            if (pathDetails.get(0) instanceof PathDetail) {
-                                for (int j = offsets[i] + 1; j < pathDetails.size(); j++) {
-                                    PathDetail pd = (PathDetail) pathDetails.get(j);
-                                    reduceLength(pathDetails, j, pd.getFirst() - removed, pd.getLast() - removed);
+                            List toSimplify = listsToSimplify.get(i);
+                            reduceLength(toSimplify, offsets[i], startIntervals[i], endIntervals[i] - removed);
+                            for (int j = offsets[i] + 1; j < toSimplify.size(); j++) {
+                                Object o = toSimplify.get(j);
+                                int first = 0;
+                                int last = 0;
+                                if (o instanceof PathDetail) {
+                                    first = ((PathDetail) o).getFirst();
+                                    last = ((PathDetail) o).getLast();
                                 }
+                                if (o instanceof Instruction) {
+                                    first = ((Instruction) o).getFirst();
+                                    last = ((Instruction) o).getLast();
+                                }
```

Comment:
here an `if&else if` might make sense with and else that throws an IllegalStateException as the list is of an unknown type.

Reply:
I actually want the Instruction and PathDetail implement the same interface, that all provide getLast, getFirst, getLength, etc. so we could solve most of this without any instanceof.

Reply:
For now I would have no problem with instanceof.